source/UF/Size-TruncatedConnected.lagda

```diff
@@ -0,0 +1,413 @@
+Ian Ray, 4th Febuary 2024.
+
+Modifications made by Ian Ray on 14 October 2024.
+
+Modifications made by Ian Ray on 7 January 2025.
+
+We develop some results that relate size, truncation and connectedness from
+the following paper:
+[1] Christensen, J.D. (2024), Non-accessible localizations.
+    Journal of Topology, 17: e12336.
+    https://doi.org/10.1112/topo.12336
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+
+module UF.Size-TruncatedConnected
+        (fe : Fun-Ext)
+       where
+
+open import MLTT.Spartan hiding (_+_)
+open import Notation.CanonicalMap
+open import Notation.Decimal
+open import UF.ConnectedTypes fe
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.PropTrunc
+open import UF.SmallnessProperties
+open import UF.Size
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Truncations fe
+open import UF.TruncationLevels
+open import UF.TruncatedTypes fe
+open import UF.Univalence
+
+module SizeTruncationandConnectedness
+        (te : general-truncations-exist)
+        (𝓥 : Universe)
+       where
+
+ private
+  pt : propositional-truncations-exist
+  pt = general-truncations-give-propositional-truncations te
+
+ open import UF.ImageAndSurjection pt
+
+\end{code}
+
+We begin by giving some definitions from [1]. We have fixed the universe parameter
+𝓥 and use it as our point of reference for 'smallness'. Also, note that we only
+use Univalence locally as some critical results hold in its absence.
+
+\begin{code}
+
+ _is_locally-small : (X : 𝓤 ̇) → (n : ℕ) → 𝓤 ⊔ (𝓥 ⁺) ̇
+ X is zero locally-small = X is 𝓥 small
+ X is (succ n) locally-small = (x x' : X) → (x ＝ x') is n locally-small
+
+ being-locally-small-is-prop : {X : 𝓤 ̇} {n : ℕ}
+                             → Univalence
+                             → is-prop (X is n locally-small)
+ being-locally-small-is-prop {_} {X} {zero} ua = being-small-is-prop ua X 𝓥
+ being-locally-small-is-prop {_} {X} {succ n} ua =
+  Π₂-is-prop fe (λ x y → being-locally-small-is-prop ua)
+
+ being-locally-small-is-upper-closed : {X : 𝓤 ̇} {n : ℕ}
+                               → X is n locally-small
```

Comment:
You should align the arrows with the colon. Please check for this elsewhere.

Reply:
Ahh yeah.